src/anomalib/data/transforms/utils.py

```diff
@@ -0,0 +1,23 @@
+"""Utility functions for working with Torchvision Transforms."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from torchvision.transforms.v2 import Compose, Transform
+
+
+def get_transforms_of_type(input_transform: Transform | None, transform_type: type[Transform]) -> list[type[Transform]]:
```

Comment:
```suggestion
def get_transforms_by_type(input_transform: Transform | None, transform_type: type[Transform]) -> list[type[Transform]]:
```

or `filter_transforms_by_type`?

Reply:
For `filter_transforms_by_type` I would expect the input to be a sequence of transform objects.

This is not entirely accurate, since the function extracts the transforms from a single other transform, which could be an actual transform or a Compose.

Maybe `extract_transforms_by_type`?